pkg/operator/ceph/cluster/mgr/mgr.go

```diff
@@ -80,31 +79,28 @@ func New(context *clusterd.Context, namespace, version string, placement rookalp
 // Start the mgr instance
 func (c *Cluster) Start() error {
 	logger.Infof("start running mgr")
+	name := instanceName()
+	if err := c.createKeyring(c.Namespace, name); err != nil {
+		return fmt.Errorf("failed to create mgr keyring. %+v", err)
+	}
 
-	for i := 0; i < c.Replicas; i++ {
-		name := fmt.Sprintf("%s%d", appName, i)
-		if err := c.createKeyring(c.Namespace, name); err != nil {
-			return fmt.Errorf("failed to create mgr keyring. %+v", err)
-		}
-
-		// start the deployment
-		deployment := c.makeDeployment(name)
-		if _, err := c.context.Clientset.ExtensionsV1beta1().Deployments(c.Namespace).Create(deployment); err != nil {
-			if !errors.IsAlreadyExists(err) {
-				return fmt.Errorf("failed to create mgr deployment. %+v", err)
-			}
-			logger.Infof("%s deployment already exists", name)
-		} else {
-			logger.Infof("%s deployment started", name)
+	// start the deployment
+	deployment := c.makeDeployment(name)
+	if _, err := c.context.Clientset.ExtensionsV1beta1().Deployments(c.Namespace).Create(deployment); err != nil {
+		if !errors.IsAlreadyExists(err) {
```

Comment:
Just to note here, these points would be perfect for automated upgrades of stateless components such as the MGR.

Reply:
not sure i understand this comment. Are you saying if it already exists, it should be updated instead of just skipping it?

Reply:
@travisn If it already exists, update it (e.g. future config changes, would also allow on the fly updates here). So the admin has less to do :)

Reply:
We need to do this as part of a more complete upgrade implementation. If we restart the mgr every time the operator runs through the orchestration it would restart unnecessarily.